packages/api/src/external/commonwell/document/document-query.ts

```diff
@@ -115,6 +120,28 @@ export async function queryAndProcessDocuments({
   }
 
   try {
+    const { organization, facility } = await getPatientDataWithSingleFacility(
+      patientParam,
+      facilityId
+    );
+
+    const patientCWData = getCWData(patientParam.data.externalData);
+    const hasNoCWStatus = !patientCWData || !patientCWData.status;
+
+    // If patient has no links in CW then we want to schedule a doc query
+    // and to prevent infinite loops we need to check if the patient has any CQ orgs
+    // as its only called the first time
```

Comment:
I don't understand this. Infinite loops? We get CQ org IDs to avoid double linking across diff HIEs.

And I think we should avoid referring to CQ on CW code and vice-versa, that's why its called "exclude list".

Reply:
Yea i removed the comment - the point i was trying to make was that if it was just checking `hasNoCWStatus` if there was an actual error with getting the CW status then this would just keep being called because 

- Schedule doc query
- Update patient (it fails to link)
- Trigger doc query
- hasNoCWStatus
- Repeat the steps above